portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/ExportResult.java

```diff
@@ -74,19 +102,26 @@ public Object getContinuationData() {
     return continuationData;
   }
 
-  /** Result types. */
+  public Throwable getThrowable() {
+    return throwable;
+  }
+
+  /**
+   * Result types.
+   */
   public enum ResultType {
     /**
      * Indicates the operation was successful and more items are available so the export should
      * continue.
      */
     CONTINUE,
-    /** Indicates the operation was successful and no more items are available. */
+    /**
+     * Indicates the operation was successful and no more items are available.
+     */
     END,
-    /** Indicates an unrecoverable error was raised. */
-    ERROR,
-
-    /** Indicates a recoverable error was raised. */
-    RETRY
+    /**
+     * Indicates an unrecoverable error was raised.
```

Comment:
does this also include the case where all retries have failed? if so, I think that should be documented, i.e. unrecoverable error or all retries failed

Reply:
oh I see. It seems odd to store the result in an ExportResult if it's a success or unrecoverable error, but not retried error. I would have ExportResult store all results (success and all failure modes).

That is also something that could be useful to store (in a db) later on.